diff --git a/plugins/FeedsProcessor.inc b/plugins/FeedsProcessor.inc
index 80e54d5..aa1ec2b 100644
--- a/plugins/FeedsProcessor.inc
+++ b/plugins/FeedsProcessor.inc
@@ -12,7 +12,7 @@ define('FEEDS_UPDATE_EXISTING', 2);
 
 // Default limit for creating items on a page load, not respected by all
 // processors.
-define('FEEDS_PROCESS_LIMIT', 50);
+define('FEEDS_PROCESS_LIMIT', 300);
 
 /**
  * Thrown if a validation fails.
