--- a/file_entity.admin.inc
+++ b/file_entity.admin.inc
@@ -372,6 +372,7 @@
       'size' => format_size($file->filesize),
       'author' => theme('username', array('account' => user_load($file->uid))),
       'timestamp' => format_date($file->timestamp, 'short'),
+      'changed' => format_date($file->changed, 'short'),
 
     );
     // Build a list of all the accessible operations for the current file.
