diff --git a/media_gallery.addimage.js b/media_gallery.addimage.js
index 2341b6e..def76cc 100644
--- a/media_gallery.addimage.js
+++ b/media_gallery.addimage.js
@@ -11,6 +11,10 @@ Drupal.media_gallery = {};
 Drupal.media_gallery.open_browser = function (event) {
   event.preventDefault();
   event.stopPropagation();
+
+  // get node id for the link that was clicked and save to settings (id attribute for parent article is node-[id])
+  Drupal.settings.mediaGalleryNodeNid = $(this).closest("article.contextual-links-region").attr('id').substring(5);
+
   var pluginOptions = { 'id': 'media_gallery', 'multiselect' : true , 'types': Drupal.settings.mediaGalleryAllowedMediaTypes};
   Drupal.media.popups.mediaBrowser(Drupal.media_gallery.add_media, pluginOptions);
 };
@@ -26,7 +30,7 @@ Drupal.media_gallery.add_media = function (mediaFiles) {
     //console.warn('Error: Media not added.');
   };
 
-  var src = Drupal.settings.mediaGalleryAddImagesUrl;
+  var src = Drupal.settings.mediaGalleryAddImagesBaseUrl + Drupal.settings.mediaGalleryNodeNid + '/' + Drupal.settings.mediaGalleryToken;
 
   var media = [];
 
diff --git a/media_gallery.dragdrop.js b/media_gallery.dragdrop.js
index 8af15cd..59794c6 100644
--- a/media_gallery.dragdrop.js
+++ b/media_gallery.dragdrop.js
@@ -3,6 +3,14 @@ Drupal.behaviors.mediaGallerySort = {};
 
 Drupal.behaviors.mediaGallerySort.attach = function (context, settings) {
   var $ = jQuery;
+  var nid = null;
+
+  // if this is a node view page, set the mediaGalleryNodeNid setting
+  if ($("body").attr("class").match("page\-node\-[^ ]+")) {
+    nid = $("body").attr("class").match("page\-node\-[^ ]+").toString().substring(10);
+    settings.mediaGalleryNodeNid = nid;
+  }
+
   // Create a drag-and-drop editor for gallery collections.
   var $collection = $('.media-gallery-collection', context).once('media-gallery-sortable');
   $('.media-collection-item-wrapper', $collection).once('media-gallery-draggable', Drupal.mediaGallerySort.addDraggableIcon);
@@ -15,7 +23,7 @@ Drupal.behaviors.mediaGallerySort.attach = function (context, settings) {
     $collection.bind('sortstop', Drupal.mediaGallerySort.setHeight);
   }
   // Create a drag-and-drop editor for individual gallery grid pages.
-  var $gallery = $('.media-gallery-view-full.media-gallery-media > .field-items').once('media-gallery-sortable');
+  var $gallery = $('article#node-' + nid + ' .media-gallery-view-full.media-gallery-media > .field-items').once('media-gallery-sortable');
   $('.media-gallery-item-wrapper', $gallery).once('media-gallery-draggable', Drupal.mediaGallerySort.addDraggableIcon);
   if ($gallery.length && settings.mediaGallerySortGalleryUrl) {
     $gallery.sortable();
@@ -47,8 +55,13 @@ Drupal.mediaGallerySort.setHeight = function (event) {
  */
 Drupal.mediaGallerySort.addDraggableIcon = function () {
   var $ = jQuery;
-  $(this).addClass('draggable');
-  $(this).prepend($('<div class="draggable-wrapper"><a class="draggable-handle">Drag</a></div>'));
+
+  // only add drag and drop functionality if this is a view node page and only apply it to that media gallery
+  var nid = $(this).closest("article.contextual-links-region").attr('id').substring(5);
+  if (Drupal.settings.mediaGalleryNodeNid != null && Drupal.settings.mediaGalleryNodeNid == nid) {
+    $(this).addClass('draggable');
+    $(this).prepend($('<div class="draggable-wrapper"><a class="draggable-handle">Drag</a></div>'));
+  }
 }
 
 /**
diff --git a/media_gallery.install b/media_gallery.install
index c3a4694..89626d1 100644
--- a/media_gallery.install
+++ b/media_gallery.install
@@ -64,25 +64,11 @@ function media_gallery_install() {
     styles_style_flush();
   }
 
-  // Add the taxonomy vocabulary for media gallery collections.
-  $vocabulary = media_gallery_create_taxonomy_vocab();
-
-  // Make sure the standard 'field_tags' field exists.
-  _media_gallery_ensure_field_tags();
-
   // Create fields (but not instances yet) for media_gallery nodes and
   // for the gallery collection vocabulary.
   foreach (_media_gallery_controlled_fields() as $field) {
     _media_gallery_ensure_field($field);
   }
-  // Attach fields to gallery_collection taxonomy terms.
-  foreach (_media_gallery_controlled_instances('taxonomy_term') as $instance) {
-    _media_gallery_ensure_instance($instance);
-  }
-
-  // Now that the gallery_collection vocabulary exists and has fields attached,
-  // create an "All galleries" term for galleries to belong to by default.
-  media_gallery_create_taxonomy_term($vocabulary);
 
   // Attach fields to the media gallery node type (including a term reference
   // for the default collection).
@@ -117,7 +103,6 @@ function media_gallery_requirements() {
   $required_fields = _media_gallery_controlled_fields();
   // In addition to the fields we control, we also need the standard field_tags
   // that most sites will have gotten from their install profile.
-  $required_fields['field_tags'] = array('type' => 'taxonomy_term_reference');
 
   foreach ($required_fields as $field_name => $field_definition) {
     $field = field_info_field($field_name);
@@ -275,18 +260,6 @@ function _media_gallery_controlled_fields() {
       'locked' => TRUE,
       'type' => 'number_integer',
     ),
-    'media_gallery_collection' => array(
-      'field_name' => 'media_gallery_collection',
-      'type' => 'taxonomy_term_reference',
-      'settings' => array(
-        'allowed_values' => array(
-          array(
-            'vocabulary' => 'gallery_collections',
-            'parent' => 0,
-          ),
-        ),
-      ),
-    ),
     // Fields to create on media items.
     'media_description' => array(
       'field_name' => 'media_description',
@@ -655,26 +628,6 @@ function _media_gallery_controlled_instances($group = NULL) {
         ),
       ),
     ),
-    // The 'collection' tag field on media gallery nodes.
-    'media_gallery_collection' => array(
-      'field_name' => 'media_gallery_collection',
-      'label' => $t('Gallery collection'),
-      'default_value' => array(
-        array(
-          'tid' => variable_get('media_gallery_default_collection_tid'),
-        ),
-      ),
-      'display' => array(
-        'default' => array(
-          'type' => 'hidden',
-          'label' => 'hidden',
-        ),
-        'teaser' => array(
-          'type' => 'hidden',
-          'label' => 'hidden',
-        ),
-      ),
-    ),
   );
   foreach ($node_instances as &$instance) {
     $instance['entity_type'] = 'node';
@@ -705,23 +658,7 @@ function _media_gallery_controlled_instances($group = NULL) {
     ),
   );
 
-  foreach ($instances as $key => $instance) {
-    // Since we are re-using fields which are defined for the node, we need to
-    // remove any additional view modes which don't belong to avoid E_NOTICE errors.
-    $instance['display'] = array_intersect_key($instance['display'], array_flip(array('default', 'full')));
-    $instance['entity_type'] = 'taxonomy_term';
-    $instance['bundle'] = 'gallery_collections';
-    $taxonomy_instances['taxo_term_' . $key] = $instance;
-  }
-
-  switch ($group) {
-    case 'node':
-      return $node_instances;
-    case 'taxonomy_term':
-      return $taxonomy_instances;
-    default:
-      return $node_instances + $taxonomy_instances;
-  }
+  return $node_instances;
 }
 
 /**
@@ -780,16 +717,6 @@ function _media_required_instances() {
         'media_gallery_detail' => array('type' => 'text_default', 'label' => 'above'),
       ),
     ),
-    'field_tags' => array(
-      'field_name' => 'field_tags',
-      'label' => $t('Tags'),
-      'widget' => array(
-        'type' => 'taxonomy_autocomplete',
-      ),
-      'display' => array(
-        'default' => array('type' => 'hidden'),
-      ),
-    ),
     'field_license' => array(
       'field_name' => 'field_license',
       'label' => $t('License settings for this media'),
@@ -1372,3 +1299,79 @@ function media_gallery_update_7200() {
   field_delete_instance($instance, TRUE);
   field_delete_field($instance['field_name']);
 }
+
+/**
+ * Renames field media_gallery_media => media_gallery_file
+ */
+function media_gallery_update_7011() {
+  $old = 'media_gallery_media';
+  $new = 'media_gallery_file';
+
+  if (db_table_exists('field_data_'.$new)) {
+    if (db_select('field_data_'.$new, 'f')->fields('f')->execute()->rowCount()==0 && db_table_exists('field_data_'.$old)) {
+      db_drop_table('field_data_'.$new); 
+      db_drop_table('field_revision_'.$new); 
+    } else {
+      return t('Media gallery already using new field.  Not performing upgrade.');
+    }
+  }
+  
+  //see: http://zgadzaj.com/how-to-change-the-machine-name-of-a-content-field-in-drupal-7
+  $transaction = db_transaction();
+  $schema = array(
+    'media_gallery_media_fid' => array(
+        //int(10) 		UNSIGNED 	Yes 	NULL
+      'type' => 'int',
+      'unsignged' => TRUE,
+      'not null' => FALSE,
+      'default' => NULL,
+    ),    
+    'media_gallery_media_display' => array(
+        //tinyint(3) 		UNSIGNED 	No 	1
+      'type' => 'int',
+      'size' => 'tiny',
+      'unsignged' => TRUE,
+      'not null' => TRUE,
+      'default' => 1,
+    ),
+    'media_gallery_media_description' => array(
+      //text 	utf8_general_ci 		Yes 	NULL
+      'type' => 'text',
+      'not null' => FALSE,
+      'default' => NULL,
+    ),
+  );
+    
+  try {
+    db_update('field_config')
+      ->fields(array('field_name' => $new))
+      ->condition('field_name', $old)
+      ->execute();
+      
+    db_update('field_config_instance')
+      ->fields(array('field_name' => $new))
+      ->condition('field_name', $old)
+      ->execute();
+    
+    foreach (array('field_data_', 'field_revision_') as $table_pre) {
+      $old_table = $table_pre . $old;
+      $new_table = $table_pre . $new;
+    
+      db_rename_table($old_table, $new_table);
+  
+      //drop index.  media_gallery_media_fid
+      db_drop_index($new_table, 'media_gallery_media_fid');  
+      foreach (array('fid', 'display', 'description') as $field) {
+        $spec = $schema["${old}_${field}"];
+        $keys = ($field == 'fid') ? array('indexes' => array('media_gallery_file_fid' => array('media_gallery_file_fid'))) : array();
+        db_change_field($new_table, "${old}_${field}", "${new}_${field}", $spec, $keys);        
+      }
+      
+    }
+  } catch (Exception $e) {
+    $transaction->rollback();
+    throw new DrupalUpdateException('Could not update Media Gallery\'s media_gallery_media field.  Rolling back changes.');
+  }
+  
+  return t('Updated Media Gallery field.');
+}
diff --git a/media_gallery.module b/media_gallery.module
index eceb654..d68f939 100644
--- a/media_gallery.module
+++ b/media_gallery.module
@@ -27,7 +27,6 @@ function media_gallery_file_view_modes() {
     'media_gallery_lightbox' => t('Gallery lightbox'),
     'media_gallery_detail' => t('Gallery detail'),
     'media_gallery_block_thumbnail' => t('Gallery block thumbnail'),
-    'media_gallery_collection_thumbnail' => t('Gallery collection thumbnail'),
   );
 }
 
@@ -74,7 +73,7 @@ function media_gallery_menu() {
   );
   // An in-gallery-context version of media/%file/edit.
   $items['media-gallery/detail/%media_gallery_mg_node/%media_gallery_mg_item/edit'] = array(
-    'title' => 'Edit file info',
+    'title' => 'Edit',
     'page callback' => 'media_gallery_media_page_edit',
     'page arguments'  => array(2, 3),
     'access callback' => 'media_gallery_edit_item_access',
@@ -86,7 +85,7 @@ function media_gallery_menu() {
     'file' => 'media_gallery.pages.inc',
   );
   $items['media-gallery/detail/%media_gallery_mg_node/%media_gallery_mg_item/remove'] = array(
-    'title' => 'Remove from gallery',
+    'title' => 'Remove',
     'page callback' => 'drupal_get_form',
     'page arguments' => array('media_gallery_remove_item_form', 2, 3),
     'access callback' => 'media_gallery_remove_item_access',
@@ -295,10 +294,7 @@ function media_gallery_view($node, $view_mode) {
         'localized_options' => array(
           'query' => array('render' => 'media-popup'),
           'attributes' => array(
-            'class' => array(
-              'media-gallery-add',
-              'launcher',
-            ),
+            'class' => 'media-gallery-add launcher',
           ),
         ),
       ),
@@ -307,11 +303,6 @@ function media_gallery_view($node, $view_mode) {
       '#suffix' => '</ul>',
     );
 
-    // Prevent the overlay module to open an additional dialog.
-    if (module_exists('overlay')) {
-      $node->content['add_media_link']['#link']['localized_options']['attributes']['class'][] = 'overlay-exclude';
-    }
-
     // Enable the "Add media" link to launch the media browser.
     module_load_include('inc', 'media', 'includes/media.browser');
     media_attach_browser_js($node->content['add_media_link']);
@@ -322,11 +313,17 @@ function media_gallery_view($node, $view_mode) {
     $instance = field_info_instance('node', 'media_gallery_file', $node->type);
     $token = drupal_get_token('media_gallery');
     $gallery_js_settings = array(
-      'mediaGalleryAddImagesUrl' => url('media-gallery/add-images/' . $node->nid . '/' . $token),
-      'mediaGallerySortGalleryUrl' => url('media-gallery/sort/gallery/' . $node->nid . '/' . $token),
+      'mediaGalleryToken' => $token,
+      'mediaGalleryNodeNid' => '',
+      'mediaGalleryAddImagesBaseUrl' => url('media-gallery/add-images/'),
       'mediaGalleryAllowedMediaTypes' => array_filter($instance['widget']['settings']['allowed_types']),
     );
 
+    // only set the js sort URL if in view node page and that node is the one being rendered
+    if (arg(1) && arg(1) == $node->nid) {
+      $gallery_js_settings['mediaGallerySortGalleryUrl'] = url('media-gallery/sort/gallery/' . $node->nid . '/' . $token);
+    }
+
     // When viewing the full node, add front-end resources for drag-and-drop
     // sorting.
     if ($view_mode == 'full') {
@@ -1279,17 +1276,6 @@ function media_gallery_contextual_links_view_alter(&$element, $items) {
       }
     }
   }
-  // Remove 'Edit' button provided by File Entity.
-  if (isset($element['#links']['media-gallery-edit']) && isset($element['#links']['file-edit'])) {
-    unset($element['#links']['file-edit']);
-  }
-  // Move 'Delete' button to the bottom of the list and modify label.
-  if (isset($element['#links']['media-gallery-edit']) && isset($element['#links']['file-delete'])) {
-    $delete = $element['#links']['file-delete'];
-    unset($element['#links']['file-delete']);
-    $element['#links']['file-delete'] = $delete;
-    $element['#links']['file-delete']['title'] = 'Delete file';
-  }
 }
 
 /**
@@ -1517,7 +1503,7 @@ function media_gallery_multiedit_access($node) {
   if ($media_items === FALSE)
     return;
 
-  $media_items = media_gallery_filter_media_access($media_items, 'update');
+  $media_items = media_gallery_filter_media_access($media_items, 'edit');
   if (count($media_items) > 0) {
     return TRUE;
   }
@@ -1554,7 +1540,7 @@ function media_gallery_view_item_access($node, $file) {
 function media_gallery_edit_item_access($node, $file) {
   // Only grant access if the user can edit the provided media
   // and the media is part of the gallery.
-  return file_entity_access('update', $file)
+  return file_entity_access('edit', $file)
     && in_array($file->fid, media_gallery_get_file_ids($node));
 }
 
@@ -1885,7 +1871,7 @@ function media_gallery_lightbox_delivery_callback($page_content) {
  * @param $media_files
  *   The files to perform the access check.
  * @param $operation
- *   A access operation (for example: 'view', 'update', 'delete', 'create').
+ *   A access operation (for example: 'view', 'edit',...).
  *
  * @return
  *   An array of files.
diff --git a/media_gallery.theme.inc b/media_gallery.theme.inc
index 56ce24c..e462b3d 100644
--- a/media_gallery.theme.inc
+++ b/media_gallery.theme.inc
@@ -232,6 +232,8 @@ function theme_media_gallery_media_item_lightbox($variables) {
   $output = 'Error';
   // If the format is to have the description as well, we add it here
   if (!empty($gallery_node->media_gallery_lightbox_extras[LANGUAGE_NONE][0]['value'])) {
+    unset($element['file']); //already rendered in media_gallery_detail.  
+
     $output =
     '<div class="mg-lightbox-wrapper clearfix">' .
       '<div class="lightbox-title">' . drupal_render($element['media_title']) . '</div>' .
