diff --git a/includes/registration.field.inc b/includes/registration.field.inc
index 41a6170..9e62145 100644
--- a/includes/registration.field.inc
+++ b/includes/registration.field.inc
@@ -226,7 +226,9 @@ function registration_field_formatter_view($entity_type, $entity, $field, $insta
             '#markup' => theme('registration_link',
               array(
                 'label' => $label,
-                'path' => $uri['path'] . '/register'
+                'path' => $uri['path'] . '/register',
+                'entity_type' => $entity_type,
+                'entity' => $entity,
               )
             )
           );
diff --git a/registration.module b/registration.module
index 32e2eb4..6659610 100644
--- a/registration.module
+++ b/registration.module
@@ -905,7 +905,7 @@ function registration_entity_settings($entity_type, $entity_id, $reset = FALSE)
 function registration_theme() {
   return array(
     'registration_link' => array(
-      'variables' => array('label' => NULL, 'path' => NULL),
+      'variables' => array('label' => NULL, 'path' => NULL, 'entity_type' => NULL, 'entity' => NULL),
     ),
     'registration_state_overview_form' => array(
       'file' => 'includes/registration.forms.inc',
