diff --git a/restful.module b/restful.module
index bee7d02..5bb2bfb 100644
--- a/restful.module
+++ b/restful.module
@@ -556,15 +556,52 @@ function restful_parse_request() {
     $request = $_POST;
   }
 
-  if (!$request && $query_string = file_get_contents('php://input')) {
-    // When trying to POST using curl on simpleTest it doesn't reach
-    // $_POST, so we try to re-grab it here.
-    // Also, sometimes the client might send the input still encoded.
-    if ($decoded_json = drupal_json_decode($query_string)) {
-      $request = $decoded_json;
-    }
-    else {
-      parse_str($query_string, $request);
+  if (!$request) {
+    $temp = explode(';', $_SERVER['CONTENT_TYPE']);
+    $contentType = $temp[0];
+    // CHARSET=<Something> will be in $temp[1], if it's ever needed
+
+    switch ($contentType) {
+      case 'application/json':
+        $request = drupal_json_decode(file_get_contents('php://input'));
+        break;
+      case 'application/x-www-form-encoded':
+        parse_str(file_get_contents('php://input'), $request);
+        break;
+      case 'multipart/form-data':
+        // php://input is an empty string when the content-type is multipart/form-data
+        // There's nothing we can do to get this data
+        throw new RestfulBadRequestException('Cannot use multipart/form-data with HTTP Method '.$method.'.');
+        break;
+      default:
+        // this is some kind of file, a direct upload
+        // copy it into the temporary directory then pass the URI to the handler for it to deal with
+        // handlers are expected to copy this file somewhere else
+        $uri = drupal_tempnam('temporary://', 'rest');
+        $input = fopen('php://input', 'rb');
+        $dest = fopen($uri, 'wb');
+
+        $size = $_SERVER['CONTENT_LENGTH'];
+        $pos = 0;
+        $chunk = 1024;
+        while ($pos < $size) {
+          if ($writ = stream_copy_to_stream($input, $dest, $chunk)) {
+            $pos += $writ;
+          } else {
+            fclose($input);
+            fclose($dest);
+            throw new RestfulUnprocessableEntityException('Unable to copy file.');
+          }
+        }
+
+        // this file never exists on the server already, as it is temporary
+        // save processing time by not checking if it already exists
+        $request['file'] = file_uri_to_object($uri, FALSE);
+        $request['file']->filemime = $contentType;
+        $request['file']->type = file_get_type($request['file']);
+        fclose($input);
+        fclose($dest);
+        break;
     }
   }
 
