diff --git a/twitter_pull.class.inc b/twitter_pull.class.inc
index ac570f7..fff7378 100644
--- a/twitter_pull.class.inc
+++ b/twitter_pull.class.inc
@@ -53,24 +53,24 @@ class twitter_puller {
       $username = drupal_substr($this->twitkey, 1, $slash - 1);
       $listname = drupal_substr($this->twitkey, $slash + 1);
       $url = 'https://api.twitter.com/1.1/lists/show.json';
-      $query = '?slug=' . urlencode($listname);
+      $query = '?slug=' . urlencode($listname) . '&tweet_mode=extended';
     }
     // if the first character is @, then consider the key a username
     elseif ($prefix == "@") {
       $key = drupal_substr($this->twitkey, 1);
       $url = 'https://api.twitter.com/1.1/statuses/user_timeline.json';
       if ($this->exclude_retweets) {
-        $query = "?screen_name=${key}&count=${num}&include_rts=false";
+        $query = "?screen_name=${key}&count=${num}&include_rts=false&tweet_mode=extended";
       }
       else {
-        $query = "?screen_name=${key}&count=${num}&include_rts=true";
+        $query = "?screen_name=${key}&count=${num}&include_rts=true&tweet_mode=extended";
       }
     }
     // if the first character is ~, then consider the key a favorites feed
     elseif ($prefix == "~") {
       $key = drupal_substr($this->twitkey, 1);
       $url = 'https://api.twitter.com/1.1/favorites/list.json';
-      $query = "?screen_name=${key}&count=${num}";
+      $query = "?screen_name=${key}&count=${num}&tweet_mode=extended";
     }
     // otherwise, use the key as a search term
     else {
@@ -139,7 +139,7 @@ class twitter_puller {
           $obj->userphoto_https = (isset($item->retweeted_status->user) && !empty($item->retweeted_status->user->profile_image_url_https)) ? $item->retweeted_status->user->profile_image_url_https : $item->retweeted_status->profile_image_url_https;
           $obj->userphoto_https = check_plain($obj->userphoto_https);
 
-          $obj->text = filter_xss($item->retweeted_status->text);
+          $obj->text = filter_xss($item->retweeted_status->full_text);
           //-- Convert date to unix timestamp so themer can easily work with it.
           $obj->timestamp = strtotime($item->retweeted_status->created_at);
 
@@ -155,7 +155,7 @@ class twitter_puller {
           $obj->userphoto_https = (isset($item->user) && !empty($item->user->profile_image_url_https)) ? $item->user->profile_image_url_https : $item->profile_image_url_https;
           $obj->userphoto_https = check_plain($obj->userphoto_https);
 
-          $obj->text = filter_xss($item->text);
+          $obj->text = filter_xss($item->full_text);
           //-- Convert date to unix timestamp so themer can easily work with it.
           $obj->timestamp = strtotime($item->created_at);
 
