--- www/includes/module.inc
+++ www/includes/module.inc
@@ -922,6 +922,9 @@
     if (strstr($hook, "access") !== FALSE) {
       1;
     }
+
+    error_log("EAM trace: " . $module . '_' . $hook);
+
     return call_user_func_array($module . '_' . $hook, $args);
   }
 }
@@ -951,6 +954,9 @@
   foreach (module_implements($hook) as $module) {
     $function = $module . '_' . $hook;
     if (function_exists($function)) {
+
+      error_log("EAM trace: " . $function);
+
       $result = call_user_func_array($function, $args);
       if (isset($result) && is_array($result)) {
         $return = array_merge_recursive($return, $result);
